binapigen/generator.go

```diff
@@ -53,39 +78,61 @@ type Generator struct {
 	messagesByName map[string]*Message
 }
 
-func New(opts Options, apiFiles []*vppapi.File, filesToGen []string) (*Generator, error) {
-	gen := &Generator{
+func (g *Generator) GetMessagesByName(name string) *Message {
+	msg, ok := g.messagesByName[name]
+	if !ok {
+		logrus.Fatalf("no message named %v found", name)
+	}
+	return msg
+}
+
+func (g *Generator) GetOpts() *Options { return &g.opts }
+
+func New(opts Options) (gen *Generator, err error) {
+	if opts.ImportPrefix == "" {
+		opts.ImportPrefix, err = ResolveImportPath(opts.OutputDir)
+		if err != nil {
+			return nil, fmt.Errorf("cannot resolve import path for output dir %s: %w", opts.OutputDir, err)
+		}
+		logrus.Debugf("resolved import path prefix: %s", opts.ImportPrefix)
+	}
+
+	apiFiles, err := vppapi.ParseDir(opts.ApiDir)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse directory %s for api files: %w", opts.ApiDir, err)
+	}
+
+	gen = &Generator{
 		FilesByName:    make(map[string]*File),
 		FilesByPath:    make(map[string]*File),
 		opts:           opts,
-		apifiles:       apiFiles,
-		filesToGen:     filesToGen,
 		enumsByName:    map[string]*Enum{},
 		aliasesByName:  map[string]*Alias{},
 		structsByName:  map[string]*Struct{},
 		unionsByName:   map[string]*Union{},
 		messagesByName: map[string]*Message{},
+		vppVersion:     vppapi.ResolveVPPVersion(opts.ApiDir),
```

Comment:
This related to the previous comment and the reasoning why the vpp version was injected into generator before.